src/DependencyInjection/Compiler/ExtensionCompilerPass.php

```diff
@@ -157,7 +137,23 @@ protected function getExtensionsForAdmin($id, Definition $admin, ContainerBuilde
      */
     protected function getManagedClass(Definition $admin, ContainerBuilder $container)
     {
-        return $container->getParameterBag()->resolveValue($admin->getArgument(1));
+        $argument = $admin->getArgument(1);
+        $class = $container->getParameterBag()->resolveValue($argument);
+
+        if (null === $class) {
+            throw new \DomainException(sprintf('The admin "%s" does not have a valid manager.', $admin->getClass()));
+        }
+
+        if (!\is_string($class)) {
+            throw new \TypeError(sprintf(
+                'Argument "%s" for admin class "%s" must be of type string, %s given.',
+                $argument,
+                $admin->getClass(),
+                \is_object($class) ? \get_class($class) : \gettype($class)
+            ));
+        }
```

Comment:
Adding these Exception/Error aren't BC-break ?

Reply:
This should not be a BC break, because the old code should have failed here if you did not provide a valid string:

https://github.com/sonata-project/SonataAdminBundle/pull/6028/files#diff-9dcbafe8f12838a40a139ccaa444c466L113

Reply:
I dont see consistency here. So this is not BC break, but this other is? 

https://github.com/sonata-project/sonata-doctrine-extensions/pull/184#discussion_r399815034

Reply:
Changed this, please review again @jordisala1991